org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/ASTConverter.java

```diff
@@ -3630,15 +3630,35 @@ public TypeParameter convert(org.eclipse.jdt.internal.compiler.ast.TypeParameter
 		return typeParameter2;
 	}
 
+	@SuppressWarnings("cast")
 	public Pattern convert(org.eclipse.jdt.internal.compiler.ast.TypePattern pattern) {
-		TypePattern typePattern = new TypePattern(this.ast);
+		TypePattern typePattern;
+		if(this.ast.apiLevel < AST.JLS22_INTERNAL) {
+			typePattern = new TypePattern(this.ast);
+		} else {
+			if(pattern.local != null && !(String.valueOf(pattern.local.name[0]) == "_") && pattern.local.type != null) { //$NON-NLS-1$
+				typePattern = new TypePattern(this.ast, true);//SingleVariableDeclaration
```

Comment:
is the comment here and below and attempt to explain something (then please elaborate), or is that some left over from experimentation (then please remove)?